### ISSUE TITLE
[Serial] Refactor `begin()` method

### ISSUE THREAD

**Issue by zfields** [ms-iot/remote-wiring#23]
Remove the random select (default constructor), then the flow becomes greatly simplified.
```
if (_deviceName) {
    listAvailableDevicesAsync();
    indentifyDeviceFromCollection();
}
connectToDeviceAsync();
```

**Comment by turkycat** [comment]
this issue is resolved in commit 7e2ba12cea2d663798f7181958e6ebca7022a6de

**Comment by turkycat** [comment]
please close this issue
